survey/forms.py

```diff
@@ -42,6 +42,10 @@ def __init__(self, *args, **kwargs):
         super(ResponseForm, self).__init__(*args, **kwargs)
         self.uuid = uuid.uuid4().hex
         self.steps_count = len(self.survey.questions.all())
+        # will contain prefetched data to avoid multiple db calls
```

Comment:
Is there a use case where the content of the database would change and we would be desynchronized?

Reply:
Just to be sure we understood the same thing of my edit: now, the response and the answers are retrieved once per page, at roughly the same time. That makes the django side consistent at a certain point of time.

I think there couldn't be a problem: the answer should not be (and cannot be through current system except through admin panel manually) separated from its response object. Therefore, if two modifications occur at the same time, the last one will override the other one which is what we want. Same with the response object, it is not really edited.

The only thing that could really happen is to have two answers for the same question and the same response. But I don't see a way to prevent that, even when forcing to retrieve answers. The only way would be to add an unique key (SQL point of view) on `Answer` (through [`unique_together`](https://docs.djangoproject.com/fr/2.2/ref/models/options/#unique-together)) on `[question, response]` but I don't know if it will not break anything else. Any idea?

Reply:
Ok, this probably never happen. Also, I don't think we can add a unique key at the SQL level because other surveys with different options can have multiple answers.